scala-package/core/src/main/scala/org/apache/mxnet/ResourceScope.scala

```diff
@@ -160,6 +162,17 @@ object ResourceScope {
     }
   }
 
+  private[mxnet] def usingIfScopeExists[A](scope: Option[ResourceScope])(body: => A): A = {
+    if (scope == None) {
+      body
+    } else {
+      ResourceScope.addToThreadLocal(scope.get)
```

Comment:
Discussed this offline so resolving.

Summary of discussion:
There current implementation of ResourceScope always uses the scope on the top of the stack as the current scope. So to get around this we are pushing the supplied scope onto the stack, running the code, then popping the scope off the stack. The alternative is to redesign ResourceScope so that it keeps track of it's current scope and it could be set. I'm creating a jira ticket to look into that possibility.